cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -70,17 +70,50 @@ func osbuildStagesToRPMs(stages []osbuild.StageResult) []koji.RPM {
 }
 
 func (impl *OSBuildJobImpl) Run(job worker.Job) error {
-	outputDirectory, err := ioutil.TempDir(impl.Output, job.Id().String()+"-*")
-	if err != nil {
-		return fmt.Errorf("error creating temporary output directory: %v", err)
+	// Initialize variables needed for reporting back to osbuild-composer
+	var outputDirectory string
+	var r []error
+	var targetResults []*target.TargetResult
+	var osbuildOutput *osbuild.Result = &osbuild.Result{
+		Success: false,
 	}
+
 	defer func() {
-		err := os.RemoveAll(outputDirectory)
+		var targetErrors []string
+		for _, err := range r {
+			errStr := err.Error()
+			fmt.Printf("target errored: %s", errStr)
+			targetErrors = append(targetErrors, errStr)
+		}
+
+		var uploadstatus string = "failure"
+		if len(targetErrors) == 0 {
+			uploadstatus = "success"
```

Comment:
Is this right even if we did not get to the uploads yet? I forgot what statuses we support, but maybe the default should be "pending" or something like that?

Reply:
hmm, in the subsequent commits I default to "failed". I think the code for each upload target should set "pending" whenever it starts the upload.

Reply:
Since the worker does not send any update to composer before the upload is finished or failed, setting `uploadstatus` to "pending" seems unnecessary to me. That is unless you also call `job.Update()`. Personally I think that defaulting to "failed" is reasonable.